in_toto/verifylib.py

```diff
@@ -240,6 +270,64 @@ def verify_layout_expiration(layout):
     raise LayoutExpiredError("Layout expired")
 
 
+def substitute_parameters(layout, parameter_dictionary):
+  """
+  <Purpose>
+    This function is a transitionary measure for parameter substitution (or
+    any other solution defined by the in-toto team). As of now, it acts as 
+    a very simple replacement layer for python-like parameters
+
+  <Arguments>
+    layout:
+            The Layout object to process.
+
+      parameter_dictionary:
+            A dictionary containing key-value pairs for substitution.
+
+  <Exceptions>
+    securesystemslib.FormatException:
+      if the parameter dictionary is malformed.
+
+  <Side Effects>
+    The layout object will have any tags replaced with the corresponding
+    values defined in the parameter dictionary.
+  """
+  _sanitize_parameter_dictionary(parameter_dictionary)
+
+  for step in layout.steps:
+
+    step.expected_command = [x.format(**parameter_dictionary) for x in step.expected_command]
+
+    new_material_rules = []
+    for rule in step.expected_materials:
+      new_rule = [x.format(**parameter_dictionary) for x in rule]
+      new_material_rules.append(new_rule)
+
+    new_product_rules = []
+    for rule in step.expected_products:
+      new_rule = [x.format(**parameter_dictionary) for x in rule]
+      new_product_rules.append(new_rule)
+
+    step.expected_materials = new_material_rules
+    step.expected_products = new_product_rules
+
+  for inspection in layout.inspect:
+
+    inspection.expected_materials = [
+        [ s.format(**parameter_dictionary) for s in m ] \
+        for m in inspection.expected_materials
+    ]
+
+    inspection.expected_products = [
+        [ s.format(**parameter_dictionary) for s in p ] \
+        for p in inspection.expected_products
+    ]
+
+    new_run = [x.format(**parameter_dictionary) for x in inspection.run]
+    inspection.run = new_run
```

Comment:
Per our guidelines against list comprehensions, please consider using s.replace() instead, and being explicit about what is being replaced.

I suggest that the `parameter_dictionary` argument be of the form:
```
{"{COMMAND}", "vim"}
```
It might be easier for others to see that what you want to replace is `{COMMAND}`  and not `COMMAND`.

Reply:
This is how it works right now. That's why we are using the built-in `format` call, rather than `replace`

Reply:
Let me use the unit test example to be clear.
```
def test_substitute(self):
"""Do a simple substitution on the expected_command field"""
substitute_parameters(self.layout, {"EDITOR": "vim"})
```

I think it might be more readable if the second argument to `substitute_parameters()` was required as follows:
```Python
{"{EDITOR}", "vim"}
```

Just a suggestion.

Reply:
Hmm, I'm not completely convinced to having the argument named like that. It just adds complexity on the codebase and I think the change in clarity is very little. We can discuss offline to decide something.